development/tools/pkg/orphanremover/remover.go

```diff
@@ -0,0 +1,313 @@
+
+	log.Print("Creating mesh of network elements to delete. This takes some time\n")
+	rawTargetPool, err := remover.computeAPI.LookupTargetPools(project)
+	if err != nil {
+		log.Fatalf("Could not list TargetPools: %v", err)
+	}
+	targetPool := extractTargetPool(rawTargetPool)
+	garbagePool := filterGarbage(targetPool, project, remover.computeAPI)
+
+	fmt.Printf("All TargetPool items: %d\n", len(targetPool))
+	fmt.Printf("Garbage TargetPool items: %d\n", len(garbagePool))
+
+	zones, err := remover.computeAPI.LookupZones(project, "europe-*")
+	if err != nil {
+		log.Fatalf("Could not list Zones: %v", err)
+	}
+	instanceGroups, err = filterInstanceGroups(zones, remover.computeAPI, project)
+	if err != nil {
+		log.Fatalf("Could not list InstanceGroups: %v", err)
+	}
+	rawURLMaps, err := remover.computeAPI.LookupURLMaps(project)
+	if err != nil {
+		log.Fatalf("Could not list UrlMaps: %v", err)
+	}
+	urlMaps := extractURLMaps(rawURLMaps)
+
+	rawBackendServices, err := remover.computeAPI.LookupBackendServices(project)
+	if err != nil {
+		log.Fatalf("Could not list BackendServices: %v", err)
+	}
+	backendServices := extractBackendServices(rawBackendServices)
+
+	rawHTTPProxies, err := remover.computeAPI.LookupHTTPProxy(project)
+	if err != nil {
+		log.Fatalf("Could not list HTTPProxy: %v", err)
+	}
+	httpProxies := extractHTTPProxies(rawHTTPProxies)
+
+	rawGlobalForwardingRules, err := remover.computeAPI.LookupGlobalForwardingRule(project)
+	if err != nil {
+		log.Fatalf("Could not list GlobalForwardingRule: %v", err)
+	}
+	globalForwardingRules := extractForwardingRules(rawGlobalForwardingRules)
+
+	remover.purge(garbagePool, instanceGroups, backendServices, urlMaps, httpProxies, globalForwardingRules, project, dryRun)
+}
+
+func (remover *Remover) purge(targetPool []targetPool, instanceGroups []instanceGroup, backendServices []backendService, urlMaps []urlMap, httpProxies []httpProxy, globalForwardingRules []globalForwardingRule, project string, dryRun bool) {
+	for _, target := range targetPool {
+		fmt.Printf("-> Processing targetPool: %s\n", target.name)
+
+		fmt.Printf("---> Delete ForwardingRules: %s in Region: %s\n", target.name, target.region)
+		if !dryRun {
+			remover.computeAPI.DeleteForwardingRule(project, target.name, target.region)
+			time.Sleep(sleepFactor * time.Second)
+		}
+		fmt.Printf("---> Delete HealthCheck: %s\n", target.healthChecks)
+		if !dryRun {
+			remover.computeAPI.DeleteHealthChecks(project, target.healthChecks)
+			time.Sleep(sleepFactor * time.Second)
+		}
+		fmt.Printf("---> Delete TargetPool: %s in Region: %s\n", target.name, target.region)
+		if !dryRun {
+			remover.computeAPI.DeleteTargetPool(project, target.name, target.region)
+			time.Sleep(sleepFactor * time.Second)
+		}
+	}
+
+	for _, group := range instanceGroups {
+		fmt.Printf("-> Processing instanceGroup: %s\n", group.name)
+		fmt.Printf("---> Delete ForwardingRules: %s\n", findGlobalForwardingRule(group.id, globalForwardingRules))
+		if !dryRun {
```

Comment:
to avoid code repetition, you can implement computeAPI interface that does nothing, and you need to replace that interface only in one place.